staging/src/k8s.io/apiserver/pkg/storage/util.go

```diff
@@ -137,7 +144,19 @@ func AnnotateInitialEventsEndBookmark(obj runtime.Object) error {
 	if objAnnotations == nil {
 		objAnnotations = map[string]string{}
 	}
-	objAnnotations["k8s.io/initial-events-end"] = "true"
+	objAnnotations[initialEventsAnnotationKey] = "true"
 	objMeta.SetAnnotations(objAnnotations)
 	return nil
 }
+
+// HasInitialEventsEndBookmarkAnnotation checks the presence of the
+// special annotation which marks that the initial events have been sent.
+func HasInitialEventsEndBookmarkAnnotation(obj runtime.Object) (bool, error) {
+	objMeta, err := meta.Accessor(obj)
+	if err != nil {
+		return false, err
+	}
+	objAnnotations := objMeta.GetAnnotations()
+	_, ok := objAnnotations[initialEventsAnnotationKey]
+	return ok, nil
```

Comment:
To be on the safe-side, should we actually check if the value is "true" ?

Reply:
I was thinking about it as well but then I realised that we didn't bother to check it for `client-go` either. 
Let me open a PR addressing it for client-go sot that we are on the safe-side.

Reply:
ptal, https://github.com/kubernetes/kubernetes/pull/120752